### PR TITLE
[NCL-4912] Add BuildTaskEndpoint implementation

### PR DESCRIPTION
**Pull request by thescouser89**
### Checklist:

* [ ] Have you added a note in the [CHANGELOG wiki](https://github.com/project-ncl/pnc/wiki/Changelog) for your change if user-facing?
* [ ] Have you added unit tests for your change?

### CODE REVIEW

**Comment by thescouser89** [comment]
💩💩💩

**Comment by thescouser89** [comment]
retest this please

**Comment by thescouser89** [comment]
Avengers: review now!

**Comment by thescouser89** [comment]
Please double check on the notification bit also!

**Comment by thescouser89** [comment]
Repushed! With the removal of 'simpleHttpPostCallback()', I also removed it in the old rest codebase.

**Comment by thescouser89** [comment]
retest this please

**Comment by janinko** [comment]
Hmm, the websockets in BPM got me thinking. The NotificationSender class is useless in new rest without moving also `rest/src/main/java/org/jboss/pnc/rest/executor/notifications/ExecutorNotificationsEndpoint.java` which contains `/ws/executor/notifications` WS endpoint.
But that endpoint is going to be removed in 2.0:

> **Honza Brazdil**: what is using the /ws/executor/notifications WS notifications?
> **Matej Lazar**: it is used to send the process progress updates: setting up env, setting up repository etc. it is not required in 2.0 as we are moving those to the logging

So please to remove the NotificationSender from the PR.

**Comment by thescouser89** [comment]
Ack

**Comment by thescouser89** [comment]
Removed ExecutorNotificationsEndpoint.java and NotificationSender

File: bpm/pom.xml [hunk 1]
```diff
@@ -74,6 +74,11 @@
       <artifactId>jboss-ejb-api_3.2_spec</artifactId>
       <scope>provided</scope>
     </dependency>
+    <dependency>
+        <groupId>org.jboss.spec.javax.websocket</groupId>
+        <artifactId>jboss-websocket-api_1.1_spec</artifactId>
+        <scope>provided</scope>
+    </dependency>
```
Comment (janinko) [inline bpm/pom.xml]: Is it stil needed?

Comment (thescouser89) [reply bpm/pom.xml]: Removed!

**Pull request merged**